utilities/logic/async_wrapper.py

```diff
@@ -170,23 +170,25 @@ def _run_module(wrapped_cmd, jid, job_path):
             sub_pid = os.fork()
             if sub_pid:
                 # the parent stops the process after the time limit
-                remaining = int(time_limit)
+                time_limit = int(time_limit)
 
                 # set the child process group id to kill all children
                 os.setpgid(sub_pid, sub_pid)
 
-                notice("Start watching %s (%s)"%(sub_pid, remaining))
-                time.sleep(step)
```

Comment:
Is it the mkdir on line 138 that we need to make sure is present? Sleep is a poor way to ensure that file systems have synced, maybe we can solve that more directly? (I am not 100% sure what the required files are in this case so not sure how we can do that yet).

That said, all we did is rely on the sleep inside of the while loop to do all the sleeping. This means that we have gone from: notice, sleep, waitpid, notice, sleep to notice, waitpid, sleep. The waitpid call should only care about the running process which is already running at this point and not care about files on disk. So I am not sure why this changes anything with respect to files.

There is a possible race if sub_pid has already exited and we will not enter the loop and sleep at all, but at that point we just exit and don't write anything to files.

All this to say I don't quite understand the concern so more details would be nice, but also take this as an argument for why I think this is fine as is (and maybe it isn't I just need more infos).